### DISCUSSION TITLE
batch_size for cluster calculation

### DISCUSSION THREAD

**Discussion by jaemolihm** [SSCHAcode/python-sscha#42] [Q&A] [answered]
Dear developers,

I'm trying to learn the SSCHA code, and I really appreciate your effort for developing and releasing this program.

I would like to ask a question about the `batch_size` parameter for a cluster calculation.

In `Cluster.py`, jobs are submitted until `count > self.batch_size`, where `count` is the number of submitted jobs (see code below). Then, in total `self.batch_size+1` jobs are submitted.
Is this an expected feature? To make at most `self.batch_size` jobs run simultaneously, one could change line 1048 of `Cluster.py` to `count >= self.batch_size:`.

```python3
            count = 0
            # Submit in parallel
            jobs = [false_id[i : i + self.job_number] for i in range(0, len(false_id), self.job_number)]
            
            for job in jobs:
                # Submit only the batch size
                if count > self.batch_size:
                    break
                t = threading.Thread(target = compute_single_jobarray, args=(job, ase_calc, ))
                t.start()
                threads.append(t)
                count += 1
```

Regards,
Jae-Mo Lihm
Seoul National University

**Comment by mesonepigreco** [comment, marked as answer]
Hi Jae-Mo Lihm,
You are right. I fix this immediately. 
Thanks for the report.
Luckily, it is not a bug that affects the calculation or the results :)
Bests,
Lorenzo
